node/silkworm/db/buffer.cpp

```diff
@@ -57,22 +53,30 @@ void Buffer::update_account(const evmc::address& address, std::optional<Account>
             bool omit_code_hash{!account_deleted};
             encoded_initial = initial->encode_for_storage(omit_code_hash);
         }
+
+        size_t payload_size{block_account_changes_.contains(block_number_) ? 0 : sizeof(BlockNum)};
         if (block_account_changes_[block_number_].insert_or_assign(address, encoded_initial).second) {
-            bump_batch_size(8, kAddressLength + encoded_initial.length());
+            payload_size += kAddressLength + encoded_initial.length();
         }
+        batch_history_size_ += payload_size;
     }
 
     if (equal) {
         return;
     }
-
-    if (accounts_.insert_or_assign(address, current).second) {
-        bump_batch_size(kAddressLength, current ? current->encoding_length_for_storage() : 0);
+    auto it{accounts_.find(address)};
+    if (it != accounts_.end()) {
+        accounts_[address] = current;
```

Comment:
Why combination of `find` & `operator[]` instead of `insert_or_assign`?

Reply:
The new version doesn't insert if address was not already in accounts_. Don't know if this was the intent.

Also why do 2 lookups and 2 constructor calls in the new version, instead of:

```
    auto it{accounts_.find(address)};
    if (it != accounts_.end()) {
        *it = current;
```

Reply:
I find the `insert_or_assign` semantic hard to digest. Checking the bool value returned due to an update (instead of new insertion) causes something like this `!accounts_.insert_or_assign(address, current).second` which always read, to me, like a failure in updating the map. But will change

Reply:
Correction ... won't change as insert_or_assign can't give me the previous value in case of update and to keep the size accounting as correct as possibile when an update occurs I substract the size of previous value and add the size of current value.

Reply:
@greg7mdp 
you're right in using the iterator but your snippet does not use it properly.
Should be `it->second = current`

Reply:
yes indeed!